### PR TITLE
Fix Accessibility COM related issues

### PR DESCRIPTION
**Pull request by JeremyKuhne**
- AccessibleObject.Navigate was not returning results from SysNavigate
- Change SysNavigate to return AccessibleObject to allow both use cases
- Update comments around SysNavigate/GetSysChild
- Add release for IDispatch in TryGetAccessibleObject when it fails
- Various other comment cleanup
- Tolerate incoming VT_INT in AccessibleObjectExtensions
- Turn ControlTabOrderHolder into record struct with clearer terms
- Make ControlTabOrderComparer static and add comments
- Use Assert.Fail for clarity in PropertyGridTests
- Add tests for AccessibleObject.Navigate

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9804)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
x86 release failed with Out of Memory, no additional info to help in the published artifacts

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjectTests.cs [hunk 1]
```diff
@@ -535,6 +535,65 @@ public void AccessibleObject_Navigate_InvokeDefault_Nop(AccessibleSelection flag
...
+    public void AccessibleObject_Navigate_FormChildren()
+    {
+        using Form form = new();
+        using Control first = new();
+        using Control second = new();
+        form.Controls.Add(first);
+        form.Controls.Add(second);
+        form.Show();
+
+        AccessibleObject firstChild = form.AccessibilityObject.Navigate(AccessibleNavigation.FirstChild);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjectTests.cs:548]: I would add this test - 
       Assert.Equal((AccessibleObject)first.TestAccessor().Dynamic.NcAccessibilityObject, firstChild);

**Review by Tanya-Solyanik — DISMISSED** [review]
Looks good, I have only cleanup-y comments.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlTabOrderHolder.cs [hunk 2]
```diff
@@ -6,21 +6,10 @@ namespace System.Windows.Forms;
...
 {
     /// <summary>
-    ///  This class contains a control and associates it with a z-order.
-    ///  This is used when sorting controls based on tab index first,
-    ///  z-order second.
+    ///  This record contains a control and associates it with a z-order.
+    ///  This is used when sorting controls based on tab index first, z-order second.
     /// </summary>
-    private class ControlTabOrderHolder
+    private readonly record struct ControlTabOrderHolder(int OriginalIndex, int TabIndex, Control? Control)
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlTabOrderHolder.cs:12]: I'm not sure that `ZOrder` is always correct so I deliberately didn't do that.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlTabOrderComparer.cs [hunk 3]
```diff
@@ -10,17 +10,24 @@ public partial class Control
...
             if (IComparerHelpers.CompareReturnIfNull(x, y, out int? returnValue))
             {
                 return (int)returnValue;
             }
 
-            int delta = x._newOrder - y._newOrder;
+            // If there is a specified tab index, use it for comparison, otherwise use the original index (which
+            // would be the index in the control collection or how Windows returns children using GW_HWNDNEXT from
+            // GW_HWNDCHILD).
+            int delta = x.TabIndex - y.TabIndex;
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlTabOrderComparer.cs:27]: Yeah, it's hard to articulate so I generalized the behavior. I didn't want to touch the logic.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjectTests.cs [hunk 4]
```diff
@@ -535,6 +535,65 @@ public void AccessibleObject_Navigate_InvokeDefault_Nop(AccessibleSelection flag
...
+    [InlineData(AccessibleNavigation.Previous, true, true)]
+    [InlineData(AccessibleNavigation.Right, false, false)]
+    [InlineData(AccessibleNavigation.Up, true, true)]
+    public void AccessibleObject_Navigate_FromForm_NoChildren(AccessibleNavigation navigate, bool returnsObject, bool isSystemAccessible)
+    {
+        using Form form = new();
+        form.Show();
+        AccessibleObject target = form.AccessibilityObject.Navigate(navigate);
+        Assert.Equal(returnsObject, target is not null);
+        if (target is not null)
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjectTests.cs:591]: I would rather check the thing I'm dereferencing.

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**